interpreter/apmint/socket.go

```diff
@@ -46,7 +47,7 @@ func openAPMAgentSocket(pid libpf.PID, socketPath string) (*apmAgentSocket, erro
 	}
 
 	// Prepend root system to ensure that this also works with containerized apps.
-	socketPath = fmt.Sprintf("/proc/%d/root/%s", pid, socketPath)
+	socketPath = path.Join("/proc/", strconv.Itoa(int(pid)), "/root/", socketPath)
```

Comment:
```suggestion
	socketPath = path.Join("/proc", strconv.Itoa(int(pid)), "root", socketPath)
```
more of question if this is more readable? `path.Join` will insert the slashes in all cases and then call Clean. Same applies to the other hunks

Reply:
Yes, we only need the leading `/` here.